create_bot_knowledge.py

```diff
@@ -80,6 +80,52 @@ def connect_to_source(project: Tuple[str, str]):
     return service, repo
 
 
+def check_directory(knowledge_dir, update_knowledge: bool):
+    if not knowledge_dir.exists():
+        if update_knowledge:
+            raise ValueError(f"No knowledge from any repo has ever been created, try using update_knowledge=False first.")
+
+        os.mkdir(knowledge_dir)
+
+def check_file(project_knowledge, update_knowledge: bool):
```

Comment:
oh yeah, i forgot to use it!
it should be used everytime, because it can happen that either

1. i don't want to update knowledge (therefore i guess that no other knowledge was ever taken so i want to create new knowledge file) but some old knowledge file already exists, so rewriting it should not be my intention?

2. i want to update knowledge but no previous knowledge was ever taken, so the update functionality does not make sense?

maybe when i re-think this, the update_knowledge should be always True, so that the process of creating knowledge would update if any files already exists, or it will just create a new file if nothing was collected before. Therefore, the parameter  would be removed.

Does this make sense?

Reply:
Yes I agree than you can remove that flag and the purpose of that function should be to check if knowledge exists or not. Then
- if the repository does not exist provide log info saying creating new knowledge for that repository and therefore considering all PR and creating the json file. 
- if the knowledge for a certain repository is already there it would log info saying it's updating the knowledge, collecting the PR and doing the difference with the one already stored and updating the json file, without recreating all json again.

Reply:
Done